sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,169 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/snapcore/snapd/sandbox/cgroup/inotify"
+)
+
+type appData struct {
```

Comment:
I'm unsure of this name, what about

```suggestion
type appMonitorData struct {
```

Also this structure could use a short doc-string explaining it's purpose

Reply:
Done.